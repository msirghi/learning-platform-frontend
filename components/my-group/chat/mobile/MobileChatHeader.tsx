import React from 'react';
import styles from '../../../../styles/modules/MyGroup.module.scss';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { IconButton } from '@material-ui/core';
import SettingsIcon from '@material-ui/icons/Settings';

type Props = {
  onBackClick: () => void;
  toggleSettings: () => void;
  online: number;
  total: number;
};

export const MobileChatHeader: React.FC<Props> = ({
  onBackClick,
  toggleSettings,
  online,
  total
}) => {
  return (
    <div className={styles.mobileHeader}>
      <div>
        <IconButton onClick={() => onBackClick()}>
          <ArrowBackIcon className={styles.backIcon} />
        </IconButton>
      </div>
      <div>
        <div className={styles.groupName}>Group name</div>
        <div className={styles.stats}>
          <span>{total} members</span>
          {online > 0 && <span> / {online} online</span>}
        </div>
      </div>
      <div className={styles.settingsIcon}>
        <IconButton onClick={() => toggleSettings()}>
          <SettingsIcon />
        </IconButton>
      </div>
    </div>
  );
};
