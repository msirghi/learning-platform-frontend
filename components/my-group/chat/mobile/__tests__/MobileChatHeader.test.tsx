import { IconButton } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { MobileChatHeader } from '../MobileChatHeader';

describe('MobileChatHeader component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <MobileChatHeader total={3} online={4} onBackClick={jest.fn()} toggleSettings={jest.fn()} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should fire prop function on arrow button click', () => {
    const spy = jest.fn();
    const wrapper = shallow(
      <MobileChatHeader total={3} online={4} onBackClick={spy} toggleSettings={jest.fn()} />
    );

    const arrowButton = wrapper.find(IconButton).at(0);
    arrowButton.simulate('click');
    expect(spy).toBeCalled();
  });

  it('should fire prop function on settings button click', () => {
    const spy = jest.fn();
    const wrapper = shallow(
      <MobileChatHeader total={3} online={4} onBackClick={jest.fn()} toggleSettings={spy} />
    );

    const settingsButton = wrapper.find(IconButton).at(1);
    settingsButton.simulate('click');
    expect(spy).toBeCalled();
  });

  it('should not render online memebers if online prop is 0', () => {
    const wrapper = shallow(
      <MobileChatHeader total={3} online={0} onBackClick={jest.fn()} toggleSettings={jest.fn()} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
