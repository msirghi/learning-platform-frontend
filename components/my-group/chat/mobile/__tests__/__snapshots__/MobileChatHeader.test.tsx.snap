// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileChatHeader component should match the snapshot 1`] = `
<div
  className="mobileHeader"
>
  <div>
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <Memo([object Object])
        className="backIcon"
      />
    </WithStyles(ForwardRef(IconButton))>
  </div>
  <div>
    <div
      className="groupName"
    >
      Group name
    </div>
    <div
      className="stats"
    >
      <span>
        3
         members
      </span>
      <span>
         / 
        4
         online
      </span>
    </div>
  </div>
  <div
    className="settingsIcon"
  >
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(IconButton))>
  </div>
</div>
`;

exports[`MobileChatHeader component should not render online memebers if online prop is 0 1`] = `
<div
  className="mobileHeader"
>
  <div>
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <Memo([object Object])
        className="backIcon"
      />
    </WithStyles(ForwardRef(IconButton))>
  </div>
  <div>
    <div
      className="groupName"
    >
      Group name
    </div>
    <div
      className="stats"
    >
      <span>
        3
         members
      </span>
    </div>
  </div>
  <div
    className="settingsIcon"
  >
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(IconButton))>
  </div>
</div>
`;
